File: sources/go/15_file_write/main.go
// task 15 file_write — expected output: 104857600
// build: go build -o prog main.go    run: ./prog
// note: plain "go build" only; do not set GOGC=off or any other tuning flags
package main

import (
	"fmt"
	"os"
)

func main() {
	buf := make([]byte, 1<<20)
	for i := range buf {
		buf[i] = byte(i % 256)
	}

	f, err := os.Create("out.bin")
	if err != nil {
		panic(err)
	}

	written := 0
	for i := 0; i < 100; i++ {
		n, err := f.Write(buf)
		if err != nil {
			panic(err)
		}
		written += n
	}

	if err := f.Sync(); err != nil {
		panic(err)
	}
	if err := f.Close(); err != nil {
		panic(err)
	}
	fmt.Println(written)
}

File: sources/kotlin/15_file_write/main.kt
// task 15 file_write — expected output: 104857600
// build: kotlinc main.kt -include-runtime -d prog.jar    run: java -jar prog.jar    (jvm row only)
// note: java.io is not available on Kotlin/Native, so this file targets the jvm row; the native row would need platform.posix write(2). Writes out.bin in 1 MiB chunks.

import java.io.FileOutputStream

fun main() {
    val chunk = 1024 * 1024
    val buf = ByteArray(chunk)
    for (i in 0 until chunk) {
        buf[i] = (i % 256).toByte()
    }
    var written = 0L
    FileOutputStream("out.bin").use { out ->
        for (t in 0 until 100) {
            out.write(buf)
            written += chunk.toLong()
        }
        out.flush()
        out.getFD().sync()
    }
    println(written)
}
